### ISSUE TITLE
cycle detected through attribute

### ISSUE THREAD

**Issue by realszopen** [MojtabaHs/iPhoneNumberField#102]
This happens on current release of iPhoneNumberField when trying to edit empty text field with placeholder text in place. It does not happen when editing text field with text already in place.

=== AttributeGraph: cycle detected through attribute 842776 ===
=== AttributeGraph: cycle detected through attribute 580900 ===
=== AttributeGraph: cycle detected through attribute 580900 ===
=== AttributeGraph: cycle detected through attribute 580900 ===
=== AttributeGraph: cycle detected through attribute 580900 ===

**Comment by keenan-chiasson** [comment]
Experiencing this as well. Any updates?

**Comment by tonybruess** [comment]
Same issue

**Comment by realszopen** [comment]
That's actually interesting. I used to experience this cycle graph in more situations. Now the number of cases seems actually smaller. Nevertheless, I have downgraded iPhoneNumberField in production app and using old version while waiting for release with fix.

**Comment by freywaid** [comment]
I would like to use this package except for these cycle errors.  Can we merge the PR that fixes this?

**Comment by MojtabaHs** [comment]
Please:
 - Fork the project
 - Add an example view into the demo project - (Think of it as a test case)
 - PR a draft request so I can investigate the root cause and merge the fix.

This way the final merged code will have a manual test file and your name as a contributer on it 🙌🏻

**Comment by MojtabaHs** [comment]
> I would like to use this package except for these cycle errors. Can we merge the PR that fixes this?

Since I can't reproduce the original issue, you may want to directly use the PR (temporary) in your project until some one is able to make a Test-Case-PR as I described above.

**Comment by realszopen** [comment]
Done PR #103

**Comment by MojtabaHs** [comment]
I can't reproduce the issue.
What I've done:
- Tap on the empty field - (nothing strange)
- Tap on the numbers - (nothing strange)
- Tap on the clear button until it is empty - (nothing strange)
- Tap on the clear button while it is already empty - (nothing strange)
- Repeat with adding `*`, `#` and `+` signs  - (nothing strange)
- Tap on the flag - (nothing strange)
- Select different countries - (nothing strange)

I have done the above steps rotating (like starting from the 4th step to the end and then from step 1 to the 4th) but again, no issues.

Please specify the exact steps to reproduce the issue on the Demo app so we will be on the same page.

ps: you can use the following to make your case run on the demo app as your previewed:
```
struct MultiplatformDemoApp: App {
    @State var number: String = ""

    var body: some Scene {
        WindowGroup {
            ContentView()
            if #available(iOS 15.0, *) {
                iPhoneNumberFieldTestView(placeholderText: "GOTACHA", mainText: $number)
            }
        }
    }
}
```

**Comment by freywaid** [comment]
I appreciate you trying to find the issue without any input from me.  The
issue arises when I'm using a binding to an optional string:

*func* ??<T>(lhs: Binding<Optional<T>>, rhs: T) -> Binding<T> {

    Binding(

        get: { lhs.wrappedValue ?? rhs },

        set: { lhs.wrappedValue = $0 }

    )

}


struct MyView: View {

    @Binding var text: String?


    var body: some View {

        iPhoneNumberField(text: $text ?? "")

    }

}
